gmso/external/convert_parmed.py

```diff
@@ -158,41 +170,65 @@ def from_parmed(structure, refer_type=True):
         top.add_connection(top_connection, update_types=False)
 
     for dihedral in structure.dihedrals:
-        # Generate dihedral parameters for DihedralType that gets passed
-        # to Dihedral
+        # Generate parameters for ImproperType or DihedralType that gets passed
+        # to corresponding Dihedral or Improper
         # These all follow periodic torsions functions
-        # (even if they are improper dihedrals)
         # Which are the default expression in top.DihedralType
         # These periodic torsion dihedrals get stored in top.dihedrals
+        # and periodic torsion impropers get stored in top.impropers
         if dihedral.improper:
```

Comment:
The parmed documentation suggests that atom3 is the central atom if this flag is set for a dihedral. Have you found that to be the case?

https://parmed.github.io/ParmEd/html/topobj/parmed.topologyobjects.Dihedral.html#parmed.topologyobjects.Dihedral

Reply:
I spot checked the top file of `124-trimethylbenzene` in Foyer OPLSAA test and its impropers have the order of `[Central, 1, 2, 3]`, and this ordered is the same when reading in using parmed (just `parmed.load_file()`). Found a comment (https://github.com/ParmEd/ParmEd/blob/b9f2955f792c03394686824519cabe030d5342b4/parmed/gromacs/gromacstop.py#L1557-L1568) so this sounds like a known bug for them and it has not been fixed. The order of atoms you found is probably what it should be if things flow downstream (you create a system and save it out to anything but a top format). I think this means the order of atom in their dihedral when `dihedral.improper == True` is just not reliable and we might need to test for when creating our own improper object (either here or in a different PR, also checking for the validity of other connection object like angle and proper dihedral). Here, I think we should make a note here about the parmed issue so we can come back to it later.